docs/user-guide/resource-watcher.md

```diff
@@ -0,0 +1,116 @@
+
+---
+
+## Creating a Watcher
+
+{% hint style="warning" %}
+### Who Can Perform This Action?
+Users need to have super-admin permission to create a watcher.
+{% endhint %}
+
+This page allows you to create a watcher to track events and run a job. It also shows the existing list of watchers (if any).
+
+1. Click **+ Create Watcher**. 
+
+    ![Figure 1: Watchers - Page](https://devtron-public-asset.s3.us-east-2.amazonaws.com/images/resource-watcher/watchers-page.jpg)
+
+2. Creating a watcher consists of 4 parts, fill all the sections one by one:
+    * [Basic Details](#basic-details)
+    * [Namespaces to Watch](#namespaces-to-watch)
+    * [Intercept Change in Resources](#intercept-change-in-resources)
+    * [Execute Runbook](#execute-runbook)
+
+    ![Figure 2: Create Watcher - Window](https://devtron-public-asset.s3.us-east-2.amazonaws.com/images/resource-watcher/create-watcher-window.jpg)
+
+### Basic Details
+
+Here, you can give a name and description to your watcher.
+
+![Figure 3: Adding Name and Description of Watcher](https://devtron-public-asset.s3.us-east-2.amazonaws.com/images/resource-watcher/basic-details.gif)
+
+### Namespaces to Watch
+
+Here, you can select the [namespaces](../reference/glossary.md#namespace) whose [Kubernetes resource](../reference/glossary.md#objects) you wish to monitor for changes. 
+
+* You can watch the namespace(s) across **All Clusters** (existing and future). 
+
+    ![Figure 4: Choosing Namespaces of all Clusters](https://devtron-public-asset.s3.us-east-2.amazonaws.com/images/resource-watcher/all-cluster.gif)
+
+* Or you can watch namespace(s) of **Specific Clusters**.
+
+    ![Figure 5: Choosing Namespaces of Specific Clusters](https://devtron-public-asset.s3.us-east-2.amazonaws.com/images/resource-watcher/specific-cluster.gif)
+
+{% hint style="warning" %}
+In both the above options, if you choose 'Specific Namespaces', you can further decide whether to track the namespaces you enter (by clicking 'Include selections') or to track the namespaces except the ones you enter (by clicking 'Exclude selections').
+{% endhint %}
+
+
+### Intercept Change in Resources
+
+Here, you can select the exact Kubernetes resource(s) you wish to track for changes (in the namespace(s) you selected in the previous step).
+
+![Figure 6: Picking Resources to Track](https://devtron-public-asset.s3.us-east-2.amazonaws.com/images/resource-watcher/intercept-changes.gif)
+
+* You can choose the resource from the **Resource kind(s) to watch** dropdown. Enter the link if it's a custom resource, for e.g., `install.istio.io/v1apha1/IstioOperator`
+
+* Choose the event type your watcher should listen to: `Created`, `Updated`, `Deleted`.
+
+* (Optional) You may also enter a [CEL expression](https://github.com/google/cel-spec/blob/master/doc/langdef.md) to catch a specific change in the resource's manifest.
```

Comment:
We highly recommend specifying the right filters to save your resources from catching and processing unwanted events